### PR TITLE
Add http bearer secret

### PR DESCRIPTION
**Pull request by NakulManchanda**
## Link to Issue
Closes: #4871 

Fast follow: #4200 

## Description of Changes
- Add new passport strategy to verify secret in Authorization header
- or read secret from payload "access_token"
- returns 401 on failure
- returns 200 on success

- Usage:

## setup
```
  initBearerAuth('cwSchedulerStrategy', CW_SCHEDULER_SECRET);
```

## secret route registration
```
registerRoute(
    router,
    'post',
    '/job/recomputeCounts',
    passport.authenticate('cwSchedulerStrategy', { session: false }),
    recomputeCounts.bind(this, models)
  );
```

## Extensibility
setup new strategy for another secret
```
  initBearerAuth('cwNewSecret', CW_NEW_SECRET);
```

## Sample Request
```
curl -H "Authorization: Bearer SECRET_TOKEN" -X POST https://commonwealth.im/api/job/recomputeCounts
# or
curl -X POST -H "Content-Type: application/json" -d '{"access_token":"SECRET_TOKEN"}' https://commonwealth.im/api/job/recomputeCounts
```

## "How We Fixed It"
- New passport strategy

## Test Plan
- added automated test to send request with or without secret

## Deployment Plan
- New environment variable `CW_SCHEDULER_SECRET` required
- We can either merge to parent PR or merge that to master and then merge it to master

## Other Considerations

-

### CODE REVIEW

**Comment by NakulManchanda** [comment]
@rbennettcw this is ready for review
- we need to add `CW_SCHEDULER_SECRET` as new secret in github environment variables

**Comment by rbennettcw** [comment]
API test is failing?

File: packages/commonwealth/server/passport/httpBearer.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import passport from 'passport';
+import { Strategy as BearerStrategy } from 'passport-http-bearer';
+
+// bearer return 401 if token is invalid
+export function initBearerAuth(secretName, secretValue) {
+  passport.use(
+    secretName,
+    new BearerStrategy((token, done) => {
```
Comment (jnaviask) [inline packages/commonwealth/server/passport/httpBearer.ts:8]: @NakulManchanda can we add a check for undefined secret values? To forbid usage of the route in the case where we forget to set the env var.

Comment (NakulManchanda) [reply packages/commonwealth/server/passport/httpBearer.ts:8]: when undefined it fails with 400 like it is right now

Comment (NakulManchanda) [reply packages/commonwealth/server/passport/httpBearer.ts:8]: let me try adding that check 
- to not register strategy when secretName or secretValue, 
- and check what happens on using authentication strategy which is not registered

Comment (NakulManchanda) [reply packages/commonwealth/server/passport/httpBearer.ts:20]: - check was already covered in `BearerStrategy` - 401 or 400 for undefined
- still added an extra check(s)

**Pull request merged**